tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <memory>
+#include <string.h>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+struct X {};
+
+template <class _Void, class Ty, class... Types>
+inline constexpr bool can_construct_at = false;
+
+template <class Ty, class... Types>
+inline constexpr bool can_construct_at<
+    std::void_t<decltype(std::construct_at(std::declval<Ty*>(), std::declval<Types>()...))>, Ty, Types...> = true;
+
+static_assert(can_construct_at<void, int>);
+static_assert(can_construct_at<void, const int>);
+static_assert(can_construct_at<void, volatile int>);
+static_assert(can_construct_at<void, const volatile int>);
+static_assert(can_construct_at<void, int, int>);
+static_assert(can_construct_at<void, const int, int>);
+static_assert(can_construct_at<void, volatile int, int>);
+static_assert(can_construct_at<void, const volatile int, int>);
+static_assert(can_construct_at<void, int, int&>);
+static_assert(can_construct_at<void, const int, int&>);
+static_assert(can_construct_at<void, volatile int, int&>);
+static_assert(can_construct_at<void, const volatile int, int&>);
+
+static_assert(!can_construct_at<void, int, X>);
+static_assert(!can_construct_at<void, X, int>);
+
+static_assert(!can_construct_at<void, int&>); // note that references can be constructed by not new'd

```

Comment:
Hmmm I was trying to put something in here that would indicate use of the new SFINAE rather than `is_constructible`... but maybe calling `is_constructible` would be correct?

Reply:
I think `can_construct_at` is very clearly not `is_constructible_v` even without the `void`.

> maybe calling `is_constructible` would be correct?

Nope: it's very intentionally "if this `new` expression is well-formed" instead of `is_constructible` because the latter requires destructibility which the former does not. (I'm not a fan of this level of constraint minimality, but that's what we standardized.)

Reply:
Replaced these reference tests with `struct indestructible`.